packages/core/src/modules/interest-group/interest-group-repository.ts

```diff
@@ -0,0 +1,55 @@
+import { InterestGroup, InterestGroupId, InterestGroupSchema, InterestGroupWrite } from "@dotkomonline/types"
+import { type Kysely, type Selectable } from "kysely"
+import { type Database } from "@dotkomonline/db"
+import { type Pageable, orderedQuery, Cursor } from "../../utils/db-utils"
+
+export interface InterestGroupRepository {
+  getById(id: InterestGroupId): Promise<InterestGroup | undefined>
+  getAll(take: number, cursor?: Cursor): Promise<InterestGroup[]>
+  create(values: InterestGroupWrite): Promise<InterestGroup>
+  update(id: InterestGroupId, values: Partial<InterestGroupWrite>): Promise<InterestGroup>
+  delete(id: InterestGroupId): Promise<void>
+}
+
+function mapToInterestGroup(interestGroup: Selectable<Database["interestGroup"]>): InterestGroup {
+  return InterestGroupSchema.parse(interestGroup)
+}
+
+export class InterestGroupRepositoryImpl implements InterestGroupRepository {
+  constructor(private readonly db: Kysely<Database>) {}
+
+  async getById(id: InterestGroupId): Promise<InterestGroup | undefined> {
+    const interestGroup = await this.db.selectFrom("interestGroup").selectAll().where("id", "=", id).executeTakeFirst()
+    return interestGroup ? mapToInterestGroup(interestGroup) : undefined
+  }
+
+  async getAll(take: number, cursor?: Cursor): Promise<InterestGroup[]> {
+    const pageable: Pageable = { take, cursor }
+    const query = orderedQuery(this.db.selectFrom("interestGroup").selectAll().limit(pageable.take), pageable.cursor)
```

Comment:
You can remive the .limit herr as orderedQuery will handle it

Reply:
refactored to use paginatedQuery instead